src/cdomains/lval.ml

```diff
@@ -533,7 +533,7 @@ struct
       match a,b with
       | `NoOffset , `NoOffset -> true
       | `Field (f1,o1), `Field (f2,o2) when f1.fname = f2.fname -> eq o1 o2
```

Comment:
Here fields are compared just by `fname`.

Maybe we should have our own `compare`/`equal` functions for comparing fields in an uniform way. Not sure if there is any module for fields where this could be.

Reply:
In both cases, the check for `fcomp.ckey` is not needed as this will be true because the variable ids compare equal. I removed the check in the other case too.